test/hashing.jl

```diff
@@ -102,9 +104,9 @@ for a in vals, b in vals
 end
 
 for a in vals
-    if a isa AbstractArray
-        @test hash(a) == hash(Array(a)) == hash(Array{Any}(a))
-    end
```

Comment:
Instead of disabling the test for `IdentityUnitRange`, can we instead check against `OffsetArray(2:4, 1)`? You can get that from `testhelpers/`, see the start of `offsetarray.jl`.

Reply:
Good call. Added in https://github.com/JuliaLang/julia/pull/35624.